pkg/operator/controller/status/controller_test.go

```diff
@@ -664,3 +664,146 @@ func Test_computeOperatorEvaluationConditionsDetectedCondition(t *testing.T) {
 		})
 	}
 }
+
+func Test_computeOperatorDegradedCondition(t *testing.T) {
+	ic := func(name string) operatorv1.IngressController {
+		return operatorv1.IngressController{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Status: operatorv1.IngressControllerStatus{
+				Conditions: []operatorv1.OperatorCondition{
+					{
+						Type:    "Degraded",
+						Status:  operatorv1.ConditionUnknown,
+						Reason:  "dummy",
+						Message: "dummy",
+					},
+				},
+			},
+		}
+	}
+	icWithStatus := func(name string, condStatus bool) operatorv1.IngressController {
+		ic := ic(name)
+		ic.Status.Conditions[0].Status = operatorv1.ConditionFalse
+		if condStatus {
+			ic.Status.Conditions[0].Status = operatorv1.ConditionTrue
+		}
+		return ic
+	}
+
+	testCases := []struct {
+		description     string
+		state           operatorState
+		expectCondition configv1.ClusterOperatorStatusCondition
+	}{
+		{
+			description: "no ingresscontrollers exist",
+			state:       operatorState{},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorDegraded,
+				Status:  configv1.ConditionTrue,
+				Reason:  "IngressDoesNotExist",
+				Message: `The "default" ingress controller does not exist.`,
+			},
+		},
+		{
+			description: "no default ingresscontroller",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("test1", false),
+					icWithStatus("test2", false),
+				},
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorDegraded,
+				Status:  configv1.ConditionTrue,
+				Reason:  "IngressDoesNotExist",
+				Message: `The "default" ingress controller does not exist.`,
+			},
+		},
+		{
+			description: "default ingresscontroller degraded",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("default", true),
+					icWithStatus("test", false),
+				},
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorDegraded,
+				Status:  configv1.ConditionTrue,
+				Reason:  "IngressDegraded",
+				Message: `The "default" ingress controller reports Degraded=True: dummy: dummy.`,
```

Comment:
Is this the revised format?
```suggestion
				Message: `The "default" ingress controller reports Degraded=True. dummy: dummy.`,
```

Reply:
I didn't change the condition message for this part, so `:` is [what we use right now](https://github.com/openshift/cluster-ingress-operator/blob/4eb3a46bb788a9066a647ddfc25792b99fd39f5a/pkg/operator/controller/status/controller.go#L445). I don't want to change this behavior because it's unrelated to this PR.